### ISSUE TITLE
Dataclass Mapping changed in sqalchemy 2.0

### ISSUE THREAD

**Issue by PaleNeutron** [agronholm/sqlacodegen#248]
Refer: https://docs.sqlalchemy.org/en/20/orm/dataclasses.html#orm-declarative-native-dataclasses

Current documented dataclass style is:

```python
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column
from sqlalchemy.orm import MappedAsDataclass


class Base(MappedAsDataclass, DeclarativeBase):
    """subclasses will be converted to dataclasses"""


class User(Base):
    __tablename__ = "user_account"

    id: Mapped[int] = mapped_column(init=False, primary_key=True)
    name: Mapped[str]
```

or at least:

```python
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column
from sqlalchemy.orm import registry


reg = registry()


@reg.mapped_as_dataclass
class User:
    __tablename__ = "user_account"

    id: Mapped[int] = mapped_column(init=False, primary_key=True)
    name: Mapped[str]
```

Columns type should be `Mapped[int]` not `int` which is different with current sqlacodegen and the new `mapped_column` is much more readable.

**Comment by agronholm** [comment]
Yeah, this needs updating. I'm not going to be working on this project for a little while (I have a queue of more urgent issues to address) so a PR would go a long way towards addressing this.

**Comment by amacfie** [comment]
Try latest `master`

**Comment by NixBiks** [comment]
`init=False` is not being set in `dataclasses` generator for automatic id's e.g. (and relations). I think that's a mistake, no?

**Comment by agronholm** [comment]
How then would you explicitly set the primary key value on insert?

**Comment by NixBiks** [comment]
True. But how would you let the server set it otherwise? You are not allowed to make an instance without the id

**Comment by agronholm** [comment]
This is a dilemma I haven't been able to figure out yet. As you probably know, SQLAlchemy really lets you do it either way, and it's just the annotations that are the issue here.

**Comment by NixBiks** [comment]
Yeah I was just trying to figure out if it's possible. Doesn't seem like it. They don't really mention anything about it in the docs.

**Comment by sheinbergon** [comment]
@agronholm looks like the main issue / feature request have already been dealt with. 
About the follow up discussion regarding setting `init=False` on autogenerated fields, I'm actually on board with @NixBiks. 
IMO conceptually speaking, meeting generated value constraints set up and maintained by the database should not be a data-class concern

**Comment by sheinbergon** [comment]
> [@agronholm](https://github.com/agronholm) looks like the main issue / feature request have already been dealt with. About the follow up discussion regarding setting `init=False` on autogenerated fields, I'm actually on board with [@NixBiks](https://github.com/NixBiks). IMO conceptually speaking, meeting generated value constraints set up and maintained by the database should not be a data-class concern

@agronholm can we discuss this?

**Comment by agronholm** [comment]
I'm fine with going with whatever the SQLAlchemy docs recommend.

**Comment by sheinbergon** [comment]
No real opinionation there, just examples portraying init=False for PK properties. This will be part of  the next milestone
